mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/JobActor.java

```diff
@@ -1614,17 +1614,19 @@ private ScheduleRequest createSchedulingRequest(
          * attributes from the job definition itself. The keys from the job definition take precedence over the
          * keys from the artifact's tags.
          *
-         * @param artifactID the artifact used by the job whose attributes are to be fetched
+         * @param artifactName the artifact used by the job whose attributes are to be fetched
          * @return A merged map of scheduling attributes. The precedence of keys follows: job definition > artifact's tags.
          */
-        private Map<String, String> mergeJobDefAndArtifactAssigmentAttributes(ArtifactID artifactID) {
+        private Map<String, String> mergeJobDefAndArtifactAssigmentAttributes(String artifactName) {
             try {
-                JobArtifact artifact = jobStore.getJobArtifact(artifactID);
+                LOGGER.info("[fdc-91] mergeJobDefAndArtifactAssigmentAttributes() artifactName --> {}   ---> {}", artifactName, artifactName.substring(0, artifactName.indexOf('.')));
+
+                JobArtifact artifact = jobStore.getJobArtifact(ArtifactID.of(artifactName.substring(0, artifactName.indexOf('.'))));
```

Comment:
I have a feeling that this is a hack, to be honest. The detail is too hidden and difficult to find.

Reply:
100% true.